pkg/virtual/framework/fixedgvs/register.go

```diff
@@ -36,14 +46,28 @@ func (vw *FixedGroupVersionsVirtualWorkspace) Register(rootAPIServerConfig gener
 			GenericConfig: &genericapiserver.RecommendedConfig{Config: *rootAPIServerConfig.Config, SharedInformerFactory: rootAPIServerConfig.SharedInformerFactory},
 			ExtraConfig: apiserver.ExtraConfig{
 				GroupVersion:    groupVersionAPISet.GroupVersion,
-				AddToScheme:     groupVersionAPISet.AddToScheme,
 				StorageBuilders: make(map[string]func(apiGroupAPIServerConfig genericapiserver.CompletedConfig) (restStorage.Storage, error)),
 			},
 		}
 		for resourceName, builder := range restStorageBuilders {
+			builder := builder
 			cfg.ExtraConfig.StorageBuilders[resourceName] = builder
 		}
 
+		scheme := runtime.NewScheme()
+		if err := kubernetesscheme.AddToScheme(scheme); err != nil {
+			return nil, err
+		}
+		if err := groupVersionAPISet.AddToScheme(scheme); err != nil {
+			return nil, err
+		}
```

Comment:
why do we need that here? Not sure I like this, a scheme that is shared more than it should be.

Reply:
Not sure I understand you concern here. It's mainly doing the same thing as it used to do, but outside from (== before) the `New` function call, in order to be able to use this scheme in the `DefaultOpenAPIConfig` creation.